### DISCUSSION TITLE
Stop pnpm from generating `pnpm-lock.yaml`

### DISCUSSION THREAD

**Discussion by PythonCoderAS** [pnpm/pnpm#4970] [Q&A] [answered]
I want to use pnpm to save space but don't want to rewrite other people's repositories to use pnpm. Is there a way I can stop the generation of a lockfile when installing? I tried `pnpm install --no-frozen-lockfile` but it didn't stop the lockfile from generating.

**Comment by zigang93** [comment]
Why you not just write pnpm-lock.yaml in your `.gitignore` if you don't push to other people's repo.

**Reply by PythonCoderAS** [reply]
This is fine for one repo but it would get annoying to have to do this for like 10-20 repos.

**Reply by zigang93** [reply]
Well, that's same annoying step if you go for other workaround for multiple repos.

Since you want to use different package maneger at your teams, you must accept the trade-off. 
I really advice that go to your team to discuss why should use pnpm over than other package manager.. 

another thing that I want to point out is pnpm default `auto-install-peers` is false, you might found out some package is missing its peer dependencies.. https://github.com/pnpm/pnpm/discussions/3995

**Reply by robations** [reply]
@PythonCoderAS You can also set global gitignore rules as well (useful for ignoring log files, .vscode, .idea directories etc).

But IMHO the package manager is the project choice, not the user's choice.

**Comment by zkochan** [comment, marked as answer]
```
pnpm install --no-lockfile
```

https://pnpm.io/npmrc#lockfile
